Website and webapp

aljazerzen opened the pull request:
- removed prql-web.
- added website
- added webapp
- migrated README content into website. Probably I moved too much - @max-sixty can you take a look on how to organize the content?

TODO:
- [x] GH actions for deployment (I left old ones in they can be removed)
- [x] figure out domains
- [ ] tasks for installing hugo

aljazerzen committed:
remove prql-web

aljazerzen committed:
add website and webapp

aljazerzen committed:
migrate README into website

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

max-sixty commented on README.md:
```diff
@@ -23,361 +23,7 @@ News](https://news.ycombinator.com/item?id=30060784#30062329) and
 [Lobsters](https://lobste.rs/s/oavgcx/prql_simpler_more_powerful_sql) earlier
 this year when it was just a proposal.
 
-## Overview
```
If we leave this in, I'm happy to do this separately and ensure there's a "Links" section at the top.

(possibly we also want to keep a small example here, I personally prefer being able to see an example when I go to repos, and I know other similar projects got flak for being too abstract in their Readmes...)

aljazerzen replied:
Yeah sure, but I think is should be shorter.

max-sixty commented on Cargo.toml:
```diff
@@ -4,7 +4,6 @@ members = [
   "prql-python",
   "prql-compiler",
   "prql-js",
-  "prql-web",
```
RIP, this was a journey. Ashes to ashes...

aljazerzen replied:
Sorry... it will remain in history and our hearts :D

max-sixty replied:
😂

max-sixty reviewed (approved):
Thanks a lot for doing the website!

- Do we need two gh-pages apps for this repo? I guess the proposed `.github` workflows won't work yet, which is fine, but when they do, will they be compatible with the language reference, which also uses gh-pages? I can have a look at this more.
- Possible we should add a license file around the logo saying it can only be used by the PRQL team (i.e. including you), rather than just anyone. I don't think we need to worry about this at stage though.

aljazerzen committed:
Merge remote-tracking branch 'origin/main' into distinct

aljazerzen committed:
prql-lang.org domain

aljazerzen committed:
guthub actions for publishing

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

aljazerzen committed:
remove gh actions for prql-web

max-sixty commented on webapp/src/workbench/prql-syntax.js:
```diff
@@ -0,0 +1,60 @@
+      [/\/\/.*$/,    'comment'],
+    ],
+  },
+};
```
Awesome, thanks for doing this!

max-sixty commented on website/content/pages/motivation.md:
```diff
@@ -0,0 +1,188 @@
+---
+title: "Motivation"
+url: motivation
+---
```
These are great. I'll do a run through of them when we merge.

max-sixty commented on website/themes/prql-theme/LICENSE:
```diff
@@ -0,0 +1,20 @@
+The MIT License (MIT)
+
+Copyright (c) 2022 PRQL core developers
```
Not important for me, but:
- Did you choose MIT above Apache? We have Apache for the rest of the repo.
- I had thought that every author retained their copyright, but licensed it under the license.

(I am fine giving copyright of my work, I don't have a preference here, I know some people might in the future — fine to push this off)

aljazerzen replied:
Whoops, that was auto-generated by hugo, will remove, so it is covered by the main LICENCE file.

Do you know, how would we add separate licences for the logo?

max-sixty replied:
> Do you know, how would we add separate licences for the logo?

I'm not sure actually; I can hopefully have a look at some point. Possibly we leave it until it matters and hope it's not too late...

max-sixty reviewed (approved):
Very nice indeed! Thank you @aljazerzen 

When we merge I'll (finally) ask the web designed to start from here.

max-sixty committed:
Merge branch 'main' into distinct

max-sixty committed:
Merge branch 'main' into distinct

max-sixty committed:
Fix tables in docs (#515)

* Fix tables in docs

Not a perfect solution by any means

max-sixty committed:
Add SQL eqivalents to transforms summary table (#516)

max-sixty committed:
Almost all the docs in a good state (#518)

* Add SQL eqivalents to transforms summary table

* Another pass of the docs

Also implementing multiline s-strings

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

max-sixty committed:
Remove 'draft' from docs (#519)

* Add SQL eqivalents to transforms summary table

* Another pass of the docs

Also implementing multiline s-strings

* Remove 'draft' from docs

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

max-sixty committed:
Refactor Syntax docs (#521)

* Refactor Syntax docs

max-sixty committed:
Add table docs (#524)

* Add table docs

aljazerzen committed:
remove LICENCE
